lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -226,7 +234,19 @@ if Enum.all?(
       {offset, state} = get_track_offset(state)
 
       track = Map.get(state.tracks, track_id)
-      track_spec = get_track_spec(offset, bin_input(pad), track, state, ctx)
+
+      remaining_tracks =
+        state.tracks
+        |> Enum.filter(fn {k, _v} -> k != track_id end)
+        |> Enum.filter(fn {_k, v} -> v.type == track.type end)
+
+      # don't link new pads if schedule_eos was send and all of the previous tracks were removed
+      track_spec =
+        if state.schedule_eos? and remaining_tracks == [] do
+          []
+        else
+          get_track_spec(offset, bin_input(pad), track, state, ctx)
+        end
```

Comment:
I thinking that raising here could be a better option instead of ignoring the new pads.

Reply:
It will raise anyway so it would be better if we do it ourselves